benchmarks/bench_sdcal.py

```diff
@@ -0,0 +1,459 @@
+
+    def _modify_tsys(self, mytsys=100.0):
+        with table_manager(self.infile, nomodify=False) as tb:
+            tsel = tb.query('SPECTRAL_WINDOW_ID IN [1,3]',
+                            sortlist='ANTENNA_ID,SPECTRAL_WINDOW_ID,TIME')
+            try:
+                nrow = tsel.nrows()
+                tsys = mytsys
+                for irow in range(nrow):
+                    tsys_spectrum = tsel.getcell('TSYS_SPECTRUM', irow)
+                    tsys_spectrum[:] = tsys
+                    tsel.putcell('TSYS_SPECTRUM', irow, tsys_spectrum)
+                    # tsys += 100.0
+            finally:
+                tsel.close()
+
+
+    # TBD
+    # @fill_weightspectrum
+    # @normal_case(interp='linear')
+    # def test_apply_sky15(self):
+    #     """test_apply_sky15 --- check if WEIGHT_SPECTRUM is updated properly when it exists."""
+    #     self.result = sdcal(infile=self.infile, calmode='apply',
+    #                         applytable=self.applytable, interp='linear')
+
+    def modify_tsys(func):
+        import functools
+
+        @functools.wraps(func)
+        def wrapper(self):
+            with table_manager(os.path.join(self.infile, 'SYSCAL'), nomodify=False) as tb:
+                tsel = tb.query('SPECTRAL_WINDOW_ID IN [1,3]',
+                                sortlist='ANTENNA_ID,SPECTRAL_WINDOW_ID,TIME')
+                try:
+                    nrow = tsel.nrows()
+                    tsys = 100.0
+                    for irow in range(nrow):
+                        tsys_spectrum = tsel.getcell('TSYS_SPECTRUM', irow)
+                        tsys_spectrum[:] = tsys
+                        tsel.putcell('TSYS_SPECTRUM', irow, tsys_spectrum)
+                        # tsys += 100.0
+                finally:
+                    tsel.close()
+            func(self)
+        return wrapper
+
+    #@normal_case()
+    def time_apply_sky(self):
+        """test_apply_composite00 --- on-the-fly application of sky table ('ps,apply')."""
+        sdcal(infile=self.infile, calmode='ps,apply')
+
+    @modify_tsys
+    #@normal_case(tsys=100.0)
+    def time_preapply_tsys(self):
+        """test_apply_composite01 --- on-the-fly application of sky table with existing Tsys table."""
+        sdcal(infile=self.infile, calmode='ps,apply', applytable=self.tsystable,
+              spwmap={1: [9], 3: [11]})
+
+    @modify_tsys
+    #@normal_case(tsys=100.0)
+    def time_apply_sky_tsys(self):
+        """test_apply_composite02 --- on-the-fly application of sky and tsys tables ('ps,tsys,apply')."""
+        sdcal(infile=self.infile, calmode='ps,tsys,apply',
+              spwmap={1: [9], 3: [11]})
+
+    @modify_tsys
+    #@normal_case(tsys=100.0)
+    def time_apply_sky(self):
+        """test_apply_composite03 --- on-the-fly application of sky table ('otfraster,apply')."""
+        sdcal(infile=self.infile, calmode='tsys,apply', applytable=self.applytable,
+              spwmap={1: [9], 3: [11]})
```

Comment:
Let's try running these test cases on their own with/without the decorator to see whether the custom setup method is properly isolate from the timing of the test cases themselves